src/test/time/clock_nanosleep/test_clock_nanosleep.rs

```diff
@@ -0,0 +1,467 @@
+    // should we restrict the tests we run?
+    let filter_shadow_passing = std::env::args().any(|x| x == "--shadow-passing");
+    let filter_libc_passing = std::env::args().any(|x| x == "--libc-passing");
+    // should we summarize the results rather than exit on a failed test
+    let summarize = std::env::args().any(|x| x == "--summarize");
+
+    let mut tests = get_tests();
+
+    if filter_shadow_passing {
+        tests.retain(|x| x.passing(TestEnvironment::Shadow));
+    }
+    if filter_libc_passing {
+        tests.retain(|x| x.passing(TestEnvironment::Libc));
+    }
+
+    test_utils::run_tests(&tests, summarize)?;
+
+    Ok(())
+}
+
+fn get_tests() -> Vec<test_utils::ShadowTest<(), anyhow::Error>> {
+    let mut tests: Vec<test_utils::ShadowTest<(), anyhow::Error>> = vec![];
+
+    // Encodes how Linux checks for invalid args.
+    let clockid_ord = 1;
+    let request_ord = 2;
+    let perm_ord = 3;
+
+    let clockids = [
+        Arg::<libc::clockid_t> {
+            value: libc::CLOCK_REALTIME,
+            validity: Validity::Valid,
+        },
+        Arg::<libc::clockid_t> {
+            value: libc::CLOCK_TAI,
+            validity: Validity::Valid,
+        },
+        Arg::<libc::clockid_t> {
+            value: libc::CLOCK_MONOTONIC,
+            validity: Validity::Valid,
+        },
+        Arg::<libc::clockid_t> {
+            value: libc::CLOCK_BOOTTIME,
+            validity: Validity::Valid,
+        },
+        Arg::<libc::clockid_t> {
+            value: libc::CLOCK_PROCESS_CPUTIME_ID,
+            validity: Validity::Valid,
+        },
+        Arg::<libc::clockid_t> {
+            value: libc::CLOCK_THREAD_CPUTIME_ID,
+            validity: Validity::Invalid(Verify {
+                ord: clockid_ord,
+                rv: libc::EINVAL,
+            }),
+        },
+        Arg::<libc::clockid_t> {
+            value: libc::CLOCK_MONOTONIC_RAW,
+            validity: Validity::Invalid(Verify {
+                ord: clockid_ord,
+                rv: libc::ENOTSUP,
+            }),
+        },
+        Arg::<libc::clockid_t> {
+            value: libc::CLOCK_REALTIME_COARSE,
+            validity: Validity::Invalid(Verify {
+                ord: clockid_ord,
+                rv: libc::ENOTSUP,
+            }),
+        },
+        Arg::<libc::clockid_t> {
+            value: libc::CLOCK_MONOTONIC_COARSE,
+            validity: Validity::Invalid(Verify {
+                ord: clockid_ord,
+                rv: libc::ENOTSUP,
+            }),
+        },
+        Arg::<libc::clockid_t> {
+            value: libc::CLOCK_REALTIME_ALARM,
+            validity: Validity::Invalid(Verify {
+                ord: perm_ord,
+                rv: libc::EPERM,
+            }),
+        },
+        Arg::<libc::clockid_t> {
+            value: libc::CLOCK_BOOTTIME_ALARM,
+            validity: Validity::Invalid(Verify {
+                ord: perm_ord,
+                rv: libc::EPERM,
+            }),
+        },
```

Comment:
`CLOCK_REALTIME_ALARM` and `CLOCK_BOOTTIME_ALARM` are actually valid clock ids for `clock_nanosleep` if the caller has the `CAP_WAKE_ALARM` capability, despite not being listed as valid in the man page. I verified this by switching these entries to `validity: Validity::Valid,` and running as root and the test passes.

So what should we do in Shadow? Should we
1. just ignore the need for the `CAP_WAKE_ALARM` and support these in Shadow, and then exclude these two clock ids from being tested when running the test on Linux since our tests are not normally run as root?
2. have Shadow return `EPERM` for these clock ids to match the case of Linux running the test without the capability?
3. Have Shadow return `ENOTSUP` for these clock ids, and then exclude the clock ids from being tested when running in Shadow?
4. Something else?